optuna/storages/_rdb/alembic/versions/v3.0.0.c.py

```diff
@@ -65,21 +85,28 @@ def upgrade():
         records = session.query(IntermediateValueModel).all()
         mapping = []
         for r in records:
-            float_type: IntermediateValueModel.FloatTypeEnum
+            value: float
             if np.isclose(r.intermediate_value, RDB_MAX_FLOAT) or np.isposinf(
                 r.intermediate_value
             ):
-                float_type = IntermediateValueModel.FloatTypeEnum.INF_POS
+                value = float("inf")
             elif np.isclose(r.intermediate_value, RDB_MIN_FLOAT) or np.isneginf(
                 r.intermediate_value
             ):
-                float_type = IntermediateValueModel.FloatTypeEnum.INF_NEG
+                value = float("-inf")
+            elif np.isnan(r.intermediate_value):
```

Comment:
Older versions of optuna silently stored `nan`s in databases. It would fail if the database did not accept it, but when the database accepts it, records containing verbose `nan` is created.

Reply:
I see 👍  Thank you for your explanation.